go-controller/pkg/util/multi_network.go

```diff
@@ -662,16 +662,30 @@ func GetPodNADToNetworkMapping(pod *kapi.Pod, nInfo NetInfo) (bool, map[string]*
 	for _, network := range allNetworks {
 		nadName := GetNADName(network.Namespace, network.Name)
 		if nInfo.HasNAD(nadName) {
+			if nInfo.IsPrimaryNetwork() {
+				return false, nil, fmt.Errorf("unexpected primary network %q pecified with a NetworkSelectionElement %+v", nInfo.GetNetworkName(), network)
+			}
 			if _, ok := networkSelections[nadName]; ok {
 				return false, nil, fmt.Errorf("unexpected error: more than one of the same NAD %s specified for pod %s",
 					nadName, podDesc)
 			}
 			networkSelections[nadName] = network
 		}
 	}
+
+	// Add the active network to the NSE map if it is configured
+	if activeNetwork != nil && activeNetwork.Name == nInfo.GetNetworkName() && (nInfo.TopologyType() == types.Layer2Topology || nInfo.TopologyType() == types.Layer3Topology) {
```

Comment:
I assume its https://github.com/ovn-org/ovn-kubernetes/pull/4459/commits/c0d160957ac3329cf78662e9febc7e5b8f9f95ac#r1648856235 right?

I think setting the dummy NSE should be a different logic from GetPodNADToNetworkMapping.. its tripping me up

Reply:
Good thing about doing it at  `GetPodNADToNetworkMapping` is that we have it common for all the places and we have the logic for handling NSE at the same place.